simulator/infra/ports/t/unit_test.cpp

```diff
@@ -22,7 +24,61 @@ namespace ports {
     
         static const int DATA_LIMIT = 5;
         static const uint64 CLOCK_LIMIT = 10;
+
+        class Logic
+        {
+            public:
+                enum
+                {
+                    DATA_FROM_A,
+                    DATA_FROM_B,
+                    DATA_INIT
+                };
+
+                enum 
+                {
+                    IS_FROM_A_READY,
+                    IS_FROM_B_READY,
+                    IS_INIT_READY,
+                    IS_STOP_READY
+                };
+
+                bool check_readiness( uint64 cycle, int check_code, bool is_ready);
+                bool check_data( uint64 cycle, int check_code, int data);
+
+            private:
+                static const int NONE = -1;
+                static const uint64 EXPECTED_MAX_CYCLE = 8;
+
+                const int data_logic[EXPECTED_MAX_CYCLE + 1][2] = 
```

Comment:
It's better to remove 1st index from initialization, so new lines can be added easily